vim-patch:8.1.1279,8.2.{360,361}

janlazo opened the pull request.

janlazo committed:
vim-patch:8.1.1279: cannot set 'spellang' to "sr@latin"

Problem:    Cannot set 'spellang' to "sr@latin". (Bojan Stipic)
Solution:   Allow using '@' in 'spellang'. (closes vim/vim#4342)
https://github.com/vim/vim/commit/9a061cb78ccbf78ec9ae454d37a49edccb4e94fc

janlazo committed:
version.c: update

vim-patch:8.0.1287: temp file used for viminfo may have wrong permissions
vim-patch:8.0.1345: race condition between stat() and open() for viminfo
vim-patch:8.0.1355: cursor keys don't work in MS-Windows console
vim-patch:8.0.1358: undercurl is not used in the terminal
vim-patch:8.0.1371: Shift-Insert doesn't always work in MS-Windows console
vim-patch:8.0.1422: no fallback to underline when undercurl is not set
vim-patch:8.2.0362: MS-Windows: channel test fails if grep is not available

janlazo committed:
vim-patch:8.2.0360: yaml files are only recognized by the file extension

Problem:    Yaml files are only recognized by the file extension.
Solution:   Check for a line starting with "%YAML". (Jason Franklin)
https://github.com/vim/vim/commit/8eab73132838e977092d7b46f70b4ecf6274fd6a

janlazo commented on ci/build.ps1:
```diff
@@ -36,7 +36,8 @@ $scoop = (New-Object System.Net.WebClient).DownloadString('https://get.scoop.sh'
 }
 
-scoop install diffutils perl
+scoop install diffutils
```
http://repo.msys2.org/msys/x86_64/diffutils-3.5-1-x86_64.pkg.tar.xz is unavailable.
pacman downloads from the same site. How are the mingw builds passing?

justinmk replied:
maybe https://github.com/lukesampson/scoop/issues/2722 is relevant

justinmk commented on ci/build.ps1:
```diff
@@ -36,7 +36,8 @@ $scoop = (New-Object System.Net.WebClient).DownloadString('https://get.scoop.sh'
 
-scoop install diffutils perl
+scoop install diffutils
+scoop install perl
```
> ci/Appveyor: install perl separately

 What's the motivation ?

janlazo replied:
Yes.
scoop can install perl but not diffutils because of the issue with msys2.

janlazo committed:
Revert "ci/Appveyor: install diffutils via scoop"

This reverts commit 4faf30de3ea72bd188c8894eb10f0c971ff28e90.

janlazo committed:
vim-patch:8.2.0361: internal error when using "0" for a callback

Problem:    Internal error when using "0" for a callback.
Solution:   Give a normal error. (closes vim/vim#5743)
https://github.com/vim/vim/commit/14e57909e662a43a42438e2701654af48af49b03

lacygoill commented:
Just for information, [8.2.0361](https://github.com/neovim/neovim/pull/11977/commits/b00650cfe909313e58aa8181474d7f016d9df94a) fixes a crash in Nvim.

justinmk commented on src/nvim/eval.c:
```diff
@@ -8114,10 +8114,16 @@ void get_system_output_as_rettv(typval_T *argvars, typval_T *rettv,
+  } else if (arg->v_type == VAR_STRING
+             && arg->vval.v_string != NULL
+             && ascii_isdigit(*arg->vval.v_string)) {
+    r = FAIL;
```
seems really specific to only check for a number rather than "is this a valid function", but oh well...

justinmk commented:
as a habit commits like https://github.com/neovim/neovim/pull/11977/commits/45b8dc0c3df861e5fa9ba67f8ac5e0993d2dba5d should say *why* we are  reverting it. But let's not block this.
